### PR TITLE
Rewrite lru_cache with maxsize=None and typed=False

### PR DESCRIPTION
**Pull request by MarcoGorelli**
xref https://github.com/asottile/pyupgrade/pull/393#discussion_r572895574 (@mxr is this what you were suggesting?)

### CODE REVIEW

File: pyupgrade/_plugins/lru_cache.py
Comment (mxr) [inline pyupgrade/_plugins/lru_cache.py:50]: I think it would be cleaner to rewrite this as `not node.args and _eligible(node.keywords)` and the implementation of `_eligible` would check the kwargs to ensure they fit the criteria for rewriting, which is something like:

1) 1 kwarg and it's `maxsize=None`
2) 2 kwargs and they're `maxsize=None, typed=False` in any order

Comment (mxr) [reply pyupgrade/_plugins/lru_cache.py:50]: There's also a broader question of whether or not we want to do this rewrite (I think it makes sense but will defer to Anthony)

**Comment by mxr** [comment]
Thanks for the PR - this is indeed what I was suggesting!

File: pyupgrade/_plugins/lru_cache.py
Comment (mxr) [reply pyupgrade/_plugins/lru_cache.py:50]:
```python

def _eligible(keywords: List[ast.keyword]) -> bool:
    if len(keywords) == 1:
        return _is_literal_kwarg(keywords[0], 'maxsize', None)
    elif len(keywords) == 2:
        return (
            (
                _is_literal_kwarg(keywords[0], 'maxsize', None) and
                _is_literal_kwarg(keywords[1], 'typed', False)
            ) or
            (
                _is_literal_kwarg(keywords[1], 'maxsize', None) and
                _is_literal_kwarg(keywords[0], 'typed', False)
            )
        )
    else:
        return False


def _is_literal_kwarg(
        keyword: ast.keyword, name: str, value: Optional[bool],
) -> bool:
    return (
        keyword.arg == name and
        isinstance(keyword.value, ast.NameConstant) and
        keyword.value.value is value
    )

```

Comment (MarcoGorelli) [reply pyupgrade/_plugins/lru_cache.py]: Indeed, that requires a couple of extra tests to get the coverage up, but I agree that it's cleaner / more readable. Thanks! I've added you as co-author to the commit

**Review by asottile — APPROVED** [review]
(will merge when the six one is ready!)

**Pull request merged**
